src/subcommand.cc

```diff
@@ -258,27 +258,26 @@ namespace	// command execution
       //////////////////////////////////////////////////////////////////////
 
       // close all *open* file descriptors
-      std::list<std::string> fdlist;
```

Comment:
Why not just `std::list<Pathname> fdlist;` and `std::stoi(fdstr.c_str());` on L275

Reply:
Does that work? I'm more of a C developer and was trying to reuse `readdir` from `src/source-download.cc` but the errors made me switch to the C way of doing it. I'll give that a shot.

Reply:
Oh wow, that does work. I'll push that version -- thanks!

Reply:
Pushed.